src/python/WMCore/Services/DBS/DBSErrors.py

```diff
@@ -58,3 +63,51 @@ class DBSReaderError(DataMgmtError):
     """
     def __init__(self, msg, **data):
         DataMgmtError.__init__(self, msg, 1002, **data)
+
+
+class DBSError():
+    """
+    DBSError provides generic interface for DBS (Go-based) errors
+    """
+    def __init__(self, ex):
+        try:
+            if hasattr(ex, "body"):
+                # case of DBSClient HTTPError
+                self.data = json.loads(ex.body)[0]
+            else:
+                self.data = json.loads(ex)[0]
+        except Exception as exp:
+            self.data = str(ex)
+
+    def getHttpCode(self):
+        """
+        :return: HTTP error code
+        """
+        if isinstance(self.data, dict):
+            return self.data['http']['code']
+        return self.data
```

Comment:
Here, I think we should not return the whole exception if `http/code` cannot be found. I would either return None, or maybe we could return the integer 0 (but None is likely a better choice).

Reply:
return type should be consistent and I'll return 0

Reply:
There are pros/cons in keeping the return data consistent here. One disadvantage is that I cannot be sure if return code 0 is actually what was returned from the server or if it's a default in case the code value cannot be found in the exception.
Anyhow, I do not have any strong preference here and 0 is good as well.